common/build.gradle

```diff
@@ -0,0 +1,60 @@
+// OMG JCX is *sooo* gay
+
+plugins {
+    // Nemerosa Versioning Plugin for the build info
+    id "net.nemerosa.versioning" version "2.14.0"
+}
+
+dependencies {
+    // Font width data (see above)
+    compile 'dumbdogdiner:mc-font-extractor:main:mojangles_width_data@json'
+
+    implementation 'com.google.code.gson:gson:2.8.6'
+}
+
+
+// Build Info
+// Set the timestamp format
+def dataBuildTimestamp = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+
+// Import the filter
+import org.apache.tools.ant.filters.ReplaceTokens
+
+// Define the map containing the tokens we want to replace
+def tokensMap = [
+        BUILDINFO_VERSION: project.rootProject.version,
+        BUILDINFO_DATEFORMAT: dataBuildTimestamp,
+        BUILDINFO_TIMESTAMP: new java.text.SimpleDateFormat(dataBuildTimestamp).format(new Date()),
+        BUILDINFO_COMMIT: versioning.info.commit,
+        BUILDINFO_BRANCH: versioning.info.branch,
+        BUILDINFO_ISDIRTY: versioning.info.dirty.toString()
+    ]
+
+// Create task to replace the tokens with their actual values
+// NOTE: At the moment this replaces tokens *globally* (format eg. @BUILDINFO_COMMIT@ in source code)
+task processSourceTokens(type: Sync) {
```

Comment:
Would it not be more sensible if this was in the root project?

Reply:
I've put it there for now as otherwise it's processing the entire project when it really only needs to process one file, and I'm not really sure about its behaviour in a root project either.

Reply:
It's just odd that this prints information shared between the entire project...

Reply:
Ah - the versioning plugin grabs information from git and gradle (our version + group is now set under allprojects {})

Reply:
So... is this slightly pointless?

Reply:
? - This is from the PR for buildinfo and embeds version + commit details into StickyAPI.java at buildtime so plugins can expose exactly what version of StickyAPI they are using

Reply:
Oh - can you leave a comment on this saying that's what this does? lul